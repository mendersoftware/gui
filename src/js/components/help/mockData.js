// Copyright 2020 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {
    hasDeviceConfig: true,
    hasMonitor: true
  },
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '2003-01-01'
  }
};
