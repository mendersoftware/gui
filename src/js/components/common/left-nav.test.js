// Copyright 2019 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
import React from 'react';

import { undefineds } from '../../../../tests/mockData';
import { render } from '../../../../tests/setupTests';
import LeftNav from './left-nav';

describe('LeftNav Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(
      <LeftNav
        sections={[
          {
            itemClass: 'test-list',
            items: [
              { exact: true, path: '/far/away', icon: <div>HEYA!</div>, style: { background: 'red' }, title: 'test1' },
              { path: '/far/away', title: 'test1' }
            ],
            title: 'test'
          }
        ]}
      />
    );
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
