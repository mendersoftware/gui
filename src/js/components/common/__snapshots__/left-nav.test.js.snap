// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftNav Component renders correctly 1`] = `
.emotion-0 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-1 {
  box-sizing: border-box;
  line-height: 48px;
  list-style: none;
  color: rgba(0, 0, 0, 0.54);
  font-family: Lato,sans-serif;
  font-weight: 500;
  font-size: 0.8125rem;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-2 {
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 11px;
  padding-bottom: 11px;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-2.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-2.Mui-selected {
  background-color: rgba(51, 122, 135, 0.08);
}

.emotion-2.Mui-selected.Mui-focusVisible {
  background-color: rgba(51, 122, 135, 0.2);
}

.emotion-2.Mui-disabled {
  opacity: 0.38;
}

.emotion-2.Mui-disabled {
  opacity: 1;
}

.emotion-2.active {
  background-color: #fff;
}

.emotion-2.leftNav.active {
  border-top: 1px solid #d4e9e7;
  border-bottom: 1px solid #d4e9e7;
}

.emotion-2.navLink,
.emotion-2.navLink .MuiListItemText-root {
  color: #969696;
}

.emotion-2.navLink .MuiListItemText-root {
  color: #969696;
}

.emotion-2.navLink.active .MuiListItemText-root {
  color: rgba(10, 10, 11, 0.78);
}

.emotion-2.active {
  background: #f7f7f7;
}

.emotion-3 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 0;
  margin-bottom: 0;
  color: rgba(10, 10, 11, 0.78);
}

.emotion-3 .MuiListItemText-primary {
  font-size: 0.8rem;
}

.emotion-3 .MuiListItemText-primary:hover {
  color: #337a87;
}

.emotion-4 {
  margin: 0;
  line-height: 1.5;
  font-family: Lato,sans-serif;
  font-weight: 400;
  font-size: 0.9286rem;
  display: block;
}

.emotion-5 {
  min-width: 56px;
  color: rgba(0, 0, 0, 0.54);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: inline-flex;
}

<ul
  class="MuiList-root MuiList-padding leftFixed emotion-0"
>
  <li
    class="MuiListSubheader-root MuiListSubheader-gutters emotion-1"
  >
    test
  </li>
  <a
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink test-list emotion-2"
    href="/far/away"
    style="background: red;"
  >
    <div
      class="MuiListItemText-root emotion-3"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-4"
      >
        test1
      </span>
    </div>
    <div
      class="MuiListItemIcon-root emotion-5"
    >
      <div>
        HEYA!
      </div>
    </div>
  </a>
  <a
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink test-list emotion-2"
    href="/further/away"
  >
    <div
      class="MuiListItemText-root emotion-3"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-4"
      >
        test1
      </span>
    </div>
  </a>
</ul>
`;
